sdk/identity/azure-identity/azure/identity/_credentials/azure_cli.py

```diff
@@ -0,0 +1,151 @@
+from .._internal import _scopes_to_resource
+
+if TYPE_CHECKING:
+    # pylint:disable=ungrouped-imports
+    from typing import Any
+
+CLI_NOT_FOUND = "Azure CLI not found on path"
+COMMAND_LINE = "az account get-access-token --output json --resource {}"
+
+# CLI's "expiresOn" is naive, so we use this naive datetime for the epoch to calculate expires_on in UTC
+EPOCH = datetime.fromtimestamp(0)
+
+
+class AzureCliCredential(object):
+    """Authenticates by requesting a token from the Azure CLI.
+
+    This requires previously logging in to Azure via "az login", and will use the CLI's currently logged in identity.
+    """
+
+    def get_token(self, *scopes, **kwargs):  # pylint:disable=no-self-use,unused-argument
+        # type: (*str, **Any) -> AccessToken
+        """Request an access token for `scopes`.
+
+        .. note:: This method is called by Azure SDK clients. It isn't intended for use in application code.
+
+        Only one scope is supported per request. This credential won't cache tokens. Every call invokes the Azure CLI.
+
+        :param str scopes: desired scopes for the token. Only **one** scope is supported per call.
+        :rtype: :class:`azure.core.credentials.AccessToken`
+
+        :raises ~azure.identity.CredentialUnavailableError: the credential was unable to invoke the Azure CLI.
+        :raises ~azure.core.exceptions.ClientAuthenticationError: the credential invoked the Azure CLI but didn't
+          receive an access token.
+        """
+
+        resource = _scopes_to_resource(*scopes)
+        output, error = _run_command(COMMAND_LINE.format(resource))
+        if error:
+            raise error
+
+        token = parse_token(output)
+        if not token:
+            sanitized_output = sanitize_output(output)
+            raise ClientAuthenticationError(message="Unexpected output from Azure CLI: '{}'".format(sanitized_output))
+
+        return token
+
+
+def parse_token(output):
+    """Parse output of 'az account get-access-token' to an AccessToken.
+
+    In particular, convert the CLI's "expiresOn" value, the string representation of a naive datetime, to epoch seconds.
+    """
+    try:
+        token = json.loads(output)
+        parsed_expires_on = datetime.strptime(token["expiresOn"], "%Y-%m-%d %H:%M:%S.%f")
+
+        # calculate seconds since the epoch; parsed_expires_on and EPOCH are naive
+        expires_on = (parsed_expires_on - EPOCH).total_seconds()
+
+        return AccessToken(token["accessToken"], int(expires_on))
+    except (KeyError, ValueError):
+        return None
+
+
+def get_safe_working_dir():
+    """Invoke 'az' from a directory on $PATH or a well-known install location, not the executing program's directory"""
+
+    path = os.environ.get("PATH")
+
+    if sys.platform.startswith("win"):
+        if path:
+            return path.split(";")[0]
+
+        # no system path; check well-known install locations
+        cli_path = "Microsoft SDKs\\Azure\\CLI2\\wbin"
```

Comment:
So if I pip install the CLI on Windows, I can't use this class?

Reply:
We only land here when the environment has no path defined. So if pip places the CLI outside the environment's path and these Program Files locations, that's correct, you can't use this class.

Reply:
discussed offline got my answer